ble_cus.c

```diff
@@ -45,6 +47,116 @@ static void on_write(ble_cus_t * p_cus, ble_evt_t const * p_ble_evt)
       p_cus->evt_handler(p_cus, &evt);
     }
   }
+
+  if (p_evt_write->handle == p_cus->time_sync_master_handles.value_handle) {
+    if (p_evt_write->data[0] == 0x6D) {
+      NRF_LOG_RAW_INFO("%08d [sync] made time sync master\n", systemTimeGetMs());
+      ts_tx_start(200);
+    } else if (p_evt_write->data[0] == 0x73) {
+      NRF_LOG_RAW_INFO("%08d [sync] made time sync slave\n", systemTimeGetMs());
+      ts_tx_stop();
+    }
+  }
+}
+
+static uint32_t mic_char_add(ble_cus_t * p_cus, const ble_cus_init_t * p_cus_init) {
+  uint32_t            err_code;
+  ble_gatts_char_md_t char_md;
+  ble_gatts_attr_md_t cccd_md;
+  ble_gatts_attr_t    attr_char_value;
+  ble_uuid_t          ble_uuid;
+  ble_gatts_attr_md_t attr_md;
+
+  memset(&char_md, 0, sizeof(char_md));
+
+  char_md.char_props.read   = 0;
+  char_md.char_props.write  = 0;
+  char_md.char_props.notify = 1; 
+  char_md.p_char_user_desc  = NULL;
+  char_md.p_char_pf         = NULL;
+  char_md.p_user_desc_md    = NULL;
+  char_md.p_cccd_md         = NULL; 
+  char_md.p_sccd_md         = NULL;
+  
+  memset(&attr_md, 0, sizeof(attr_md));
+
+  attr_md.read_perm  = p_cus_init->custom_value_char_attr_md.read_perm;
+  attr_md.write_perm = p_cus_init->custom_value_char_attr_md.write_perm;
+  attr_md.vloc       = BLE_GATTS_VLOC_STACK;
+  attr_md.rd_auth    = 1;
+  attr_md.wr_auth    = 1;
+  attr_md.vlen       = 1;
+
+  ble_uuid.type = p_cus->uuid_type;
+  ble_uuid.uuid = MIC_CHAR_UUID;
+
+  memset(&attr_char_value, 0, sizeof(attr_char_value));
+
+  attr_char_value.p_uuid    = &ble_uuid;
+  attr_char_value.p_attr_md = &attr_md;
+  attr_char_value.init_len  = NRF_SDH_BLE_GATT_MAX_MTU_SIZE;
+  attr_char_value.init_offs = 0;
+  attr_char_value.max_len   = NRF_SDH_BLE_GATT_MAX_MTU_SIZE;
+
+  err_code = sd_ble_gatts_characteristic_add(p_cus->service_handle, &char_md,
+                                              &attr_char_value,
+                                              &p_cus->mic_value_handles);
+  if (err_code != NRF_SUCCESS)
+  {
+      return err_code;
+  }
+
+  return NRF_SUCCESS;
+}
+
+static uint32_t time_sync_master_char_add(ble_cus_t * p_cus, const ble_cus_init_t * p_cus_init) {
+  uint32_t            err_code;
+  ble_gatts_char_md_t char_md;
+  ble_gatts_attr_md_t cccd_md;
+  ble_gatts_attr_t    attr_char_value;
+  ble_uuid_t          ble_uuid;
+  ble_gatts_attr_md_t attr_md;
+
+  memset(&char_md, 0, sizeof(char_md));
+
+  char_md.char_props.read   = 0;
+  char_md.char_props.write_wo_resp  = 1;
+  char_md.char_props.notify = 0; 
+  char_md.p_char_user_desc  = NULL;
+  char_md.p_char_pf         = NULL;
+  char_md.p_user_desc_md    = NULL;
+  char_md.p_cccd_md         = NULL; 
+  char_md.p_sccd_md         = NULL;
```

Comment:
would it be a lot of effort to convert this to the wrapper API I used that nordic provides? no worries if it's a lot of effort